tests/feedback.test.ts

```diff
@@ -0,0 +1,147 @@
+import { validate } from 'class-validator';
+import { plainToClass } from 'class-transformer';
+import { DatabaseConnection, UserFactory } from './data';
+import { FeedbackFactory } from './data/FeedbackFactory';
+import { FeedbackControllerMock } from './controllers';
+import { ActivityScope, ActivityType, FeedbackStatus, UserAccessType } from '../types';
+import { ActivityModel } from '../models/ActivityModel';
+import { FeedbackModel } from '../models/FeedbackModel';
+import { Feedback } from '../api/validators/FeedbackControllerRequests';
+import { UserModel } from '../models/UserModel';
+import { Config } from '../config';
+
+beforeAll(async () => {
+  await DatabaseConnection.connect();
+});
+
+beforeEach(async () => {
+  await DatabaseConnection.clear();
+});
+
+afterAll(async () => {
+  await DatabaseConnection.clear();
+  await DatabaseConnection.close();
+});
+
+describe('feedback submission', () => {
+  test('returns proper feedback response on successful submission', async () => {
+    const [user] = UserFactory.create(1);
+    const [feedback] = FeedbackFactory.create(1);
+
+    await FeedbackControllerMock.mock([user]);
+    const submittedFeedback = await FeedbackControllerMock.submitFeedback(feedback, user);
+
+    expect(submittedFeedback.user).toStrictEqual(user.getPublicProfile());
+    expect(submittedFeedback.status).toStrictEqual(FeedbackStatus.SUBMITTED);
+    expect(submittedFeedback).toMatchObject(feedback);
+  });
+
+  test('is invalidated when submission description is too short', async () => {
+    const [feedback] = FeedbackFactory.with({ description: 'A short description' });
+
+    const errors = await validate(plainToClass(Feedback, feedback));
+
+    expect(errors).toBeDefined();
+    expect(errors).toHaveLength(1);
+    expect(errors[0].property).toEqual('description');
+    expect(errors[0].constraints.minLength).toBeDefined();
+  });
+
+  test('persists submission data and activity in database on successful submission', async () => {
+    const [user] = UserFactory.create(1);
+    const [feedback] = FeedbackFactory.create(1);
+
+    await FeedbackControllerMock.mock([user]);
+    const submittedFeedback = await FeedbackControllerMock.submitFeedback(feedback, user);
+
+    const conn = await DatabaseConnection.get();
+    const [persistedFeedback] = await conn.manager.find(FeedbackModel, { relations: ['user'] });
+    const activity = await conn.manager.find(ActivityModel, { relations: ['user'] });
+    const feedbackSubmissionActivity = activity[1];
+
+    expect(submittedFeedback).toStrictEqual(persistedFeedback.getPublicFeedback());
```

Comment:
Same thing here with matching the entire object once instead of separate assertions for each property.